common/stringop.c

```diff
@@ -69,26 +69,22 @@ int lenient_strcmp(char *a, char *b) {
 }
 
 list_t *split_string(const char *str, const char *delims) {
-	list_t *res = create_list();
+	list_t *res = list_new(sizeof(char *), 0);
 	char *copy = strdup(str);
 	char *token;
 
 	token = strtok(copy, delims);
 	while(token) {
 		token = strdup(token);
-		list_add(res, token);
+		list_add(res, &token);
 		token = strtok(NULL, delims);
 	}
 	free(copy);
 	return res;
 }
 
 void free_flat_list(list_t *list) {
-	int i;
-	for (i = 0; i < list->length; ++i) {
-		free(list->items[i]);
-	}
-	list_free(list);
```

Comment:
If you're gutting free_flat_list, just drop it and replace its usages with `list_free_withp`.